octoprint_findmymrbeam/__init__.py

```diff
@@ -366,52 +381,39 @@ def _get_netconnectd_state(self):
 		return res
 
 	def _get_internal_modes(self):
-		internal_modes = []
-		try:
-			pluginInfo = self._plugin_manager.get_plugin_info("mrbeam")
-			if pluginInfo is not None:
-				if pluginInfo.implementation.support_mode:
-					internal_modes.append(MODE_SUPPORT)
-				if pluginInfo.implementation.calibration_tool_mode:
-					internal_modes.append(MODE_CALIBRATION_TOOL)
-		except Exception as e:
-			self._logger.exception(
-				"Exception while reading support mode state from mrbeam: {}".format(e)
-			)
-		return internal_modes
-
-	def _perform_update_request(self, uuid, scheme, port, path, search_id=None, http_user=None, http_password=None):
+		if self._internal_modes is None:
+			try:
+				internal_modes = []
+				pluginInfo = self._plugin_manager.get_plugin_info("mrbeam")
+				if pluginInfo is not None:
+					if pluginInfo.implementation.support_mode:
+						internal_modes.append(MODE_SUPPORT)
+					if pluginInfo.implementation.calibration_tool_mode:
+						internal_modes.append(MODE_CALIBRATION_TOOL)
+					self._internal_modes = internal_modes
+			except Exception as e:
+				self._logger.exception(
+					"Exception while reading support mode state from mrbeam: {}".format(e)
+				)
+		return self._internal_modes
+
+	def _get_plugin_version(self):
+		if self._mrb_plugin_version is None:
+			try:
+				pluginInfo = self._plugin_manager.get_plugin_info("mrbeam")
+				# if pluginInfo is not None:
+				# 	version = pluginInfo.version
+				if pluginInfo.implementation._plugin_version:
+					self._mrb_plugin_version = pluginInfo.implementation._plugin_version
+			except Exception as e:
+				self._logger.exception(
+					"Exception while reading version from mrbeam: {}".format(e)
+				)
+		return self._mrb_plugin_version
+
+	def _perform_update_request(self):
```

Comment:
Too many `if ... is None` and `try: ... except ...`

```py
from itertools import compress
modes = (pluginInfo.implementation.support_mode, pluginInfo.implementation.calibration_tool_mode,)
# return a tuple of activated modes like support or calibration tool
return tuple(compress((MODE_SUPPORT, MODE_CALIBRATION_TOOL), modes))
```
These functions are too simple to warrant the `self._internal_modes` caching.